s3/client.bal

```diff
@@ -61,7 +62,11 @@ public isolated client class Client {
         check generateSignature(self.accessKeyId, self.secretAccessKey, self.region, GET, SLASH, UNSIGNED_PAYLOAD,
             requestHeaders);
         http:Response httpResponse = check self.amazonS3->get(SLASH, requestHeaders);
-        xml xmlPayload = check httpResponse.getXmlPayload();
+        string stringPayload = check httpResponse.getTextPayload();
```

Comment:
This is not required here. The hanging issue can occur only when you return the same response object - `httpResponse`